examples/ssh/opkssh/cli.go

```diff
@@ -0,0 +1,308 @@
+		if len(row) > 1 {
+			email := row[0]
+			allowedPrincipals := row[1:]
+			return email, allowedPrincipals, nil
+		}
+	}
+	return "", nil, fmt.Errorf("policy file contained no policy")
+}
+
+func (p *SimpleFilePolicyEnforcer) CheckPolicy(principalDesired string, pkt *pktoken.PKToken) error {
+	allowedEmail, allowedPrincipals, err := p.ReadPolicyFile()
+	if err != nil {
+		return err
+	}
+	email, err := pkt.GetClaim("email")
+	if err != nil {
+		return err
+	}
+	if string(email) == allowedEmail {
+		if slices.Contains(allowedPrincipals, principalDesired) {
+			// Access granted
+			return nil
+		} else {
+			return fmt.Errorf("no policy to allow %s to assume %s, check policy config in %s", email, principalDesired, p.PolicyFilePath)
+		}
+	} else {
+		return fmt.Errorf("no policy for email %s, allowed email is %s, check policy config in %s", email, allowedEmail, p.PolicyFilePath)
+	}
+}
+
+func CreateSSHCert(client *parties.OpkClient, signer crypto.Signer, alg jwa.KeyAlgorithm, gqFlag bool, principals []string) ([]byte, []byte, error) {
+	pkt, err := client.OidcAuth(signer, alg, map[string]any{}, gqFlag)
+	cert, err := sshcert.BuildPktSshCert(pkt, principals)
+	if err != nil {
+		return nil, nil, err
+	}
+	caSigner, err := ssh.NewSignerFromSigner(signer)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	mas, err := ssh.NewSignerWithAlgorithms(caSigner.(ssh.AlgorithmSigner), []string{ssh.KeyAlgoECDSA256})
+	if err != nil {
+		return nil, nil, err
+	}
+
+	sshcert, err := cert.SignCert(mas)
+
+	if err != nil {
+		return nil, nil, err
+	}
+	certBytes := ssh.MarshalAuthorizedKey(sshcert)
+
+	seckeySsh, err := ssh.MarshalPrivateKey(signer, "openpubkey cert")
+	if err != nil {
+		return nil, nil, err
+	}
+	seckeySshBytes := pem.EncodeToMemory(seckeySsh)
+
+	return certBytes, seckeySshBytes, nil
+}
+
+func WriteKeys(seckeyPath string, pubkeyPath string, seckeySshPem []byte, certBytes []byte) error {
+	// Write ssh secret key to filesystem
+	err := os.WriteFile(seckeyPath, seckeySshPem, 0600)
+	if err != nil {
+		return err
+	}
+	certBytes = append(certBytes, []byte(" "+"openpubkey")...)
+
+	// Write ssh public key (certificate) to filesystem
+	err = os.WriteFile(pubkeyPath, certBytes, 0777)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func FileExists(fPath string) bool {
+	_, error := os.Open(fPath)
+	return !errors.Is(error, os.ErrNotExist)
+}
+
+func WriteKeysToSSHDir(seckeySshPem []byte, certBytes []byte) error {
+	homePath, err := os.UserHomeDir()
+	if err != nil {
+		return err
+	}
+	sshPath := filepath.Join(homePath, ".ssh")
+
+	//  To enable ssh to automatically find the key created by openpubkey we
+	// need to use a default ssh key path. However this means that this
+	// filename might already be in use by the user. To ensure we don't
+	// overwrite a ssh key not created by openpubkey we check the comment in the
+	// key to see if it was created by openpubkey
+	defaultKeyNames := []string{"id_ecdsa_sk", "id_ecdsa", "id_dsa"}
```

Comment:
Unfortunately if you include opk, then ssh will find it. ssh has a set of seven known key file names that it will find. I've selected three of them that match our algorithm. 

Anyone that wants to use ssh default key file names has to either use one of the seven and take actions to ensure they don't collide or use ssh agent. Most just fight over `id_rsa` since that it the default filename that is sent first by ssh.

Reply:
but you can solve this with an ssh config, right? Is there a reason we don't want to modify that?

Reply:
you can ask the user to modify how ssh looks up its keys by wildcarding the identity configuration. The right way to solve this is either with default files or the agent.